protocols/rendezvous/tests/rendezvous.rs

```diff
@@ -173,11 +173,19 @@ async fn discover_allows_for_dial_by_peer_id() {
     bob.behaviour_mut()
         .discover(Some(namespace.clone()), None, None, roberts_peer_id);
 
-    assert_behaviour_events! {
-        alice: rendezvous::client::Event::Registered { .. },
-        bob: rendezvous::client::Event::Discovered { .. },
-        || { }
-    };
+    // TODO: Clean up
+    match alice.select_next_some().await {
+        SwarmEvent::Behaviour(rendezvous::client::Event::Registered { .. }) => {}
+        _ => todo!(),
+    }
+
+    // TODO: Clean up
+    match bob.select_next_some().await {
+        SwarmEvent::Behaviour(rendezvous::client::Event::Discovered { registrations, .. }) => {
+            assert!(!registrations.is_empty());
+        }
+        _ => todo!(),
+    }
```

Comment:
If I am not mistaken the feature of this pull request - concurrent dialing - uncovers a race condition in this test, namely that `bob` tries to discover `alice` before `alice` registered with the rendezvous server.

@thomaseizinger suggestion on how to proceed in a way that is idiomatic with the remainder of the test harness? Maybe a flag to `assert_behaviour_events` that allows awaiting the two events either in parallel or in sequence?